pubtools/_pulp/tasks/set_maintenance/set_maintenance_off.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import re
+
+from .base import SetMaintenance
+
+
+LOG = logging.getLogger("set-maintenance")
+
+
+class SetMaintenanceOff(SetMaintenance):
+    """Unset repositories maintenance mode.
+
+    See "pub maintenance-on --help" for more information on maintenance mode.
+    """
+
+    def run(self):
+        report = self.pulp_client.get_maintenance_report().result()
+
+        to_remove = []
+        if self.args.repo_regex:
```

Comment:
I'd like to suggest putting all this "get repos" logic into a separate method and marking it with a ``@step`` decorator.  Getting/adjusting/uploading the maintenance mode possibly should be a ``@step`` as well.

I think we should try to use steps for just about every task, because it's a prerequisite for tasks to display meaningfully in the planned progress UI using the machine-readable logs.

Reply:
Makes sense.